kernel/src/scan/mod.rs

```diff
@@ -459,9 +459,9 @@ pub fn transform_to_logical(
         let result = engine
             .get_expression_handler()
             .get_evaluator(
-                read_schema.clone(),
+                read_schema,
                 read_expression.clone(),
-                DataType::Struct(Box::new(global_state.logical_schema.clone())),
+                DataType::Struct(Box::new((*global_state.logical_schema).clone())),
```

Comment:
```suggestion
                DataType::Struct(Box::new(global_state.logical_schema.as_ref().clone())),
```

We should maybe not require a fully owned `StructType` here, but for now this is fine

Reply:
I've left the fully owned struct, but changed the dereference.